### PR TITLE
[BugFix] Fix Insert overwrite memory leak on follower node (#17556)

### LINKED ISSUE #17555: [BUG] Load Job memory leak on non-leader node
### Steps to reproduce the behavior (Required)
1. Execute insert overwrite on a 3 fe cluster.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by gengjun-git**
Signed-off-by: gengjun-git <gengjun@starrocks.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #17555
The callback should be removed when replaying end load job.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by gengjun-git** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadManager.java [hunk 1]
```diff
@@ -213,6 +213,9 @@ public void replayEndLoadJob(LoadJobFinalOperation operation) {
             return;
         }
         job.unprotectReadEndOperation(operation);
+
+        Catalog.getCurrentGlobalTransactionMgr().getCallbackFactory().removeCallback(job.id);
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadManager.java:217]: The alignment is not right.

**Review by trueeyu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadManager.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadManager.java:217]: > The alignment is not right.

why

**Comment by gengjun-git** [comment]
@mergify rebase

**Comment by gengjun-git** [comment]
run starrocks_fe_unittest

**Comment by gengjun-git** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**